components/StoreUser.coffee

```diff
@@ -17,11 +17,15 @@ downloadAvatar = (avatarUrl, callback) ->
     callback new Error 'Avatar request failed'
   req.send()
 
-cleanUpUrl = (callback) ->
-  regex = new RegExp "(?:\\/)?\\?code=[A-Za-z0-9]+"
-  newLoc = window.location.href.replace regex, ''
-  if newLoc isnt window.location.href
-    window.location.href = newLoc
+cleanUpUrl = (out, callback) ->
+  [url, query] = window.location.href.split '?'
+  return callback() unless query
```

Comment:
There are two situations where we need to store user:

* Right after login, when we need to remove grant code from URL afterwards (since otherwise reloading app would cause errors since the code is stale)
* At later app loads if Flowhub API gives updated user information (for example, changed plan). In this case there is no code to clear